src/doc/doc-builders.js

```diff
@@ -1,5 +1,21 @@
 "use strict";
 
+/**
+ * @typedef {Object} DocObject
+ * @property {string} type
+ * @property {boolean} [hard]
+ * @property {boolean} [literal]
+ *
+ * @typedef {string | DocObject} Doc
+ *
+ * @typedef {Object} GroupOpts
```

Comment:
I don't think we need a typedef for this type as we are not going to refer it by name anywhere else. That's why better to move its property definitions to `@param` of the `group` function.

Reply:
As for `id`, define it as `any` and optional.

Reply:
See the syntax for for one-off type specifications in `@param` [here](https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html#typedef-callback-and-param).

Reply:
> As for `id`, define it as `any` and optional.

+1, thanks.

> See the syntax for for one-off type specifications in `@param` [here](https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html#typedef-callback-and-param).

Thanks for the pointer. But I would actually downvote the one-off syntax for a couple of reasons:

- seems to be more ugly than the typedef that I proposed
- Considering that GroupOpts is defined in the documentation, I would really favor "formally" documenting it in the source code in order to show the level of consistency

Reply:
Seriously, the name `GroupOpts` won't be used anywhere else. It's a one-off type. It doesn't need an alias.

Reply:
It's a standard practice for options objects, which are a basically how named parameters look in JS. Defining it outside the function's jsdoc block makes little sense to me.

Reply:
Would you mind if we just revert this change now, and do the one-off type in a separate PR?

Reply:
Okay, but I don't really understand this decision.